app/controllers/routes_controller.rb

```diff
@@ -11,9 +11,13 @@ def show
 
   def update
     route_details = @request_data[:route]
-    @route = Route.find_or_initialize_by(incoming_path: route_details.delete(:incoming_path))
-    status_code = @route.new_record? ? 201 : 200
-    @route.update_attributes(route_details) || status_code = 422
+    @route = Route.new(route_details)
+    if @route.valid?
```

Comment:
Oh yes, good point. I think I'm trying to guard against something else in the app Mongoid directly rather than through `upsert_on_path`, but maybe that's not necessary.